src/wasm/wasm-debug.cpp

```diff
@@ -99,26 +103,318 @@ void dumpDWARF(const Module& wasm) {
 //     StringMap<std::unique_ptr<MemoryBuffer>>
 //     EmitDebugSections(llvm::DWARFYAML::Data &DI, bool ApplyFixups);
 //
-// For modifying data, like line numberes, we can in theory do that either on
-// the DWARFContext or DWARFYAML::Data; unclear which is best, but modifying
-// the DWARFContext may save us doing fixups in EmitDebugSections.
-//
 
-void writeDWARFSections(Module& wasm) {
+// Represents the state when parsing a line table.
+struct LineState {
+  uint32_t addr = 0;
+  // TODO sectionIndex?
+  uint32_t line = 1;
+  uint32_t col = 0;
+  uint32_t file = 1;
+  // TODO uint32_t isa = 0;
+  // TODO Discriminator = 0;
+  bool isStmt;
+  bool basicBlock = false;
+  // XXX these two should be just prologue, epilogue?
+  bool prologueEnd = false;
+  bool epilogueBegin = false;
+
+  LineState() = default;
+  LineState(const LineState& other) = default;
```

Comment:
I guess if we have an explicitly-defaulted copy constructor we should have an explicitly-defaulted assignment operator too?

Reply:
It wouldn't be used yet - not sure what the benefit is to adding one at this point? Is it easier to read for you that way?

Reply:
It was used, in that compound assignment I complained about. More generally it's just a rule of thumb that whatever you do with the copy constructor (default, delete, etc) you should also do with the assignment operator.

Reply:
Ok, sounds good, added.